providers/openmetrics/client_metrics.go

```diff
@@ -0,0 +1,187 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	openmetrics "github.com/prometheus/client_golang/prometheus"
+)
+
+// ClientMetrics represents a collection of metrics to be registered on a
+// Prometheus metrics registry for a gRPC client.
+type ClientMetrics struct {
+	clientRegister openmetrics.Registerer
```

Comment:
We can remove this (:

Reply:
We are using `clientRegister` in `Enable...` functions. Should I modify the `Enable..` functions to accept clientRegister?

Reply:
Hmmm
need to look on this, happy to propose something on top of your PR (:

Reply:
Ideally register is only needed on start!